lib/AST/NestedCondProp.cpp

```diff
@@ -12,106 +12,208 @@
 #include <gflags/gflags.h>
 #include <glog/logging.h>
 
+#include <unordered_map>
 #include <vector>
 
 #include "rellic/AST/ASTBuilder.h"
 #include "rellic/AST/Util.h"
 
 namespace rellic {
-using ExprVec = std::vector<clang::Expr*>;
+// Stores a set of expression that have a known value, so that they can be
```

Comment:
Just a thought, but do you think `z3::expr::substitute` could simplify this pass? I have doubts it would, but worth a question.

Reply:
I tried to use it, and it didn't work for reasons I can't remember right now and should have probably documented 🤦 One problem I remember is that it didn't give an indication of whether it actually did anything, which meant you had to check for equality every time, which is slow as molasses

Reply:
> it didn't give an indication of whether it actually did anything
True. Defintiely not worth it then .